src/xrpld/app/tx/detail/LoanBrokerCoverWithdraw.cpp

```diff
@@ -0,0 +1,220 @@
+#include <xrpld/app/tx/detail/LoanBrokerCoverWithdraw.h>
+//
+#include <xrpld/app/misc/LendingHelpers.h>
+#include <xrpld/app/tx/detail/Payment.h>
+
+namespace ripple {
+
+bool
+LoanBrokerCoverWithdraw::checkExtraFeatures(PreflightContext const& ctx)
+{
+    return checkLendingProtocolDependencies(ctx);
+}
+
+NotTEC
+LoanBrokerCoverWithdraw::preflight(PreflightContext const& ctx)
+{
+    if (ctx.tx[sfLoanBrokerID] == beast::zero)
+        return temINVALID;
+
+    auto const dstAmount = ctx.tx[sfAmount];
+    if (dstAmount <= beast::zero)
+        return temBAD_AMOUNT;
+
+    if (!isLegalNet(dstAmount))
+        return temBAD_AMOUNT;
+
+    if (auto const destination = ctx.tx[~sfDestination])
+    {
+        if (*destination == beast::zero)
+        {
+            return temMALFORMED;
+        }
+    }
+
+    return tesSUCCESS;
+}
+
+TER
+LoanBrokerCoverWithdraw::preclaim(PreclaimContext const& ctx)
+{
+    auto const& tx = ctx.tx;
+
+    auto const account = tx[sfAccount];
+    auto const brokerID = tx[sfLoanBrokerID];
+    auto const amount = tx[sfAmount];
+
+    auto const dstAcct = tx[~sfDestination].value_or(account);
+
+    auto const sleBroker = ctx.view.read(keylet::loanbroker(brokerID));
+    if (!sleBroker)
+    {
+        JLOG(ctx.j.warn()) << "LoanBroker does not exist.";
+        return tecNO_ENTRY;
+    }
+    if (account != sleBroker->at(sfOwner))
+    {
+        JLOG(ctx.j.warn()) << "Account is not the owner of the LoanBroker.";
+        return tecNO_PERMISSION;
+    }
+    auto const vault = ctx.view.read(keylet::vault(sleBroker->at(sfVaultID)));
+    if (!vault)
+        return tefBAD_LEDGER;  // LCOV_EXCL_LINE
```

Comment:
Should probably also have a fatal log here, similar to LoanBrokerCoverDeposit

```suggestion
    {
        JLOG(ctx.j.fatal()) << "Vault is missing for Broker " << brokerID;
        return tefBAD_LEDGER;
    }
```

Reply:
> Should probably also have a fatal log here, similar to LoanBrokerCoverDeposit

Will fix.